### ISSUE TITLE
Update:1 prototype

### ISSUE THREAD

**Issue by old-droid** [old-droid/XTRAIN#1]
Its made with PyTorch #***RAPTORRRR*** (1.5)

!
import numpy as np
import pandas as pd
import torch

from datasets import load_dataset

ds = load_dataset("ggml-org/ci")

ds.shape

print(ds)

class Feed_Forward(torch.nn.Module):
  def __init__(self, d_model, d_ff):
    super().__init__()
    self.Linear_1 = torch.nn.Linear(d_model, d_ff)

    self.relu = torch.nn.ReLU()

    self.Linear_2 = torch.nn.Linear(d_ff, d_model)

  def forward(self, x):
    x = self.Linear_1(x)
    x = self.relu(x)
    x = self.Linear_2(x)

    return x

class Attention_Mechanism(torch.nn.Module):
  def __init__(self, d_model):
    super().__init__()
    self.d_k = d_model

    self.query_proj = torch.nn.Linear(d_model, d_model)
    self.key_proj = torch.nn.Linear(d_model, d_model)
    self.value_proj = torch.nn.Linear(d_model, d_model)

  def forward(self, x):

    Q = self.query_proj(x)
    K = self.key_proj(x)
    V = self.value_proj(x)

    attention_scores = torch.matmul(Q, K.transpose(-2, -1))

    scaled_scores = attention_scores / torch.sqrt(torch.tensor(self.d_k, dtype=torch.float32))

    atten_weights = torch.nn.functional.softmax(scaled_scores, dim=-1)

    context_vector = torch.matmul(atten_weights, V)

    return context_vector

class Transformer_Block(torch.nn.Module):
  def __init__(self, d_model, d_ff):
    super().__init__()
    self.attention = Attention_Mechanism(d_model)
    self.feed_forward = Feed_Forward(d_model, d_ff)

  def forward(self, x):

    atten_x = self.attention(x)
    ff_x = self.feed_forward(atten_x)

    return ff_x

**Comment by old-droid** [comment]
[copy_of_gan_draft1.py](https://github.com/user-attachments/files/23825895/copy_of_gan_draft1.py)
